### PR TITLE
chore: integrate circomspect to catch bugs in circuits

### LINKED ISSUE #4: Add Circomspect to circom circuits
**Description**

Circomspect is a static analyzer and linter for the Circom zero-knowledge DSL: https://github.com/trailofbits/circomspect. It helps identify potential issues on ZK-Kit circuits.

Resources: 
https://blog.trailofbits.com/2022/09/15/it-pays-to-be-circomspect/

### PR DESCRIPTION
**Pull request by cedoor**
## Description

This PR integrates [Circomspect](https://github.com/trailofbits/circomspect), a static analyzer and linter for the [Circom](https://iden3.io/circom) to help devs to identify potential issues before publishing code.

## Related Issue(s)

Closes #4

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [x] My changes generate no new warnings
-   [x] I have run `yarn format`, `yarn lint`, and `yarn compile` without getting any errors
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by cedoor** [comment]
@0xjei @ctrlc03 

There are some warnings that need to be checked.

File: .github/workflows/main.yml
Comment (sripwoud) [inline .github/workflows/main.yml:1]: **not blocking merge:**
I think we want to run the `style` and `test` jobs on changes in `packages/**/*.{circom,json,ts}`.
As here the same paths should trigger all the jobs to run, there is not benefit in using `tj-actions/changed-files@v44`.

So we could simplify this workflow and use the `on.<event>.paths` field to define the trigger paths instead.

**Review by sripwoud — APPROVED** [review]
tACK (ran `yarn lint` locally)

circomspect reports some `WARNING`.
We should either address them,
or accept them and use the `--allow` CLI flag

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
         "version:release": "changelogithub",
         "format": "prettier -c .",
         "format:write": "prettier -w .",
+        "lint": "circomspect packages/*/src -L ./node_modules/circomlib/circuits",
```
Comment (sripwoud) [inline package.json:17]: `circomspect` seems to return a non zero code even if only outputs of `WARNING` level are detected.
This is what the ci check fails.

So we definitely need to address the warning or allow them with eg:

```commandline
circomspect -a CS0006 -a CS007 -a CS0010 -a CS0014 -a CS0017 -a CS0018 packages/*/src -L ./node_modules/circomlib/circuits
```

use `circomspect -v` to see rule IDS.

**Review by sripwoud — CHANGES_REQUESTED** [review]

**Comment by ctrlc03** [comment]
> @0xjei @ctrlc03
> 
> There are some warnings that need to be checked.

Am I supposed to see those warning here? can't see anything

**Comment by sripwoud** [comment]
@ctrlc03 I think you have to run `yarn lint` locally to see them :wink: 

Later we could the `--sarif` CLI flag to upload the results to GitHub so you can see them as comments in the PR. (like we do with slither in the zk-kit.solidity repo)

**Comment by sripwoud** [comment]
@cedoor check https://github.com/privacy-scaling-explorations/zk-kit.circom/pull/8

File: package.json (hunk 1 above)
Comment (sripwoud) [reply package.json:17]: see #8 for a fix

Comment (cedoor) [reply package.json:17]: Thanks!

**Review by sripwoud — APPROVED** [review]

**Comment by 0xjei** [comment]
Since this has been merged, should we take care of the warnings in another PR? @sripwoud @cedoor

**Comment by cedoor** [comment]
@0xjei @ctrlc03 please sync your main branch, install `circomspect`, and run `yarn lint` to see warnings in your circuits!

**Pull request merged**